Docs Audit #2 (#3801)
dfinity/portal: docs/developer-docs/smart-contracts/maintain/control.mdx

# Control

<MarkdownChipRow labels={["Beginner", "Concept"]} />

## Overview

A <GlossaryTooltip>canister</GlossaryTooltip> is managed by a list of controllers. A controller is specified by a <GlossaryTooltip>principal</GlossaryTooltip>, which can be self-authenticating, e.g. a `dfx` developer identity, or another canister. Canisters can have multiple controllers where each has the same administrative rights, or it can have no controller, in which case the canister becomes immutable (blackholed) and cannot be upgraded or deleted.

## Setting the controllers of a canister

When a canister is created, the default controller list contains the developer identity used to create the canister. Additional controllers can be specified during canister creation or they can be added in the future.

> To specify controllers upon canister creation, the [`dfx canister create --controller`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-create) command can be used.

Author: ```suggestion
To specify controllers upon canister creation, the [`dfx canister create --controller <controller>`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-create) command can be used.
```

> To specify controllers upon canister creation, the [`dfx canister create --controller <controller>`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-create) command can be used.

Reviewer: ```suggestion
To specify a different controller upon canister creation, the [`dfx canister create --controller <controller>`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-create) command can be used.
```

Reviewer: if multiple controllers can be defined, this should be improved in the `dfx` outputs

> To update a canister's controllers after the canister has been created, the [`dfx canister update-settings`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-update-settings) command can be used with the `--add-controller` or `--remove-controller` flag to add or remove controllers respectively.

Author: ```suggestion
To update a canister's controllers after the canister has been created, the [`dfx canister update-settings`](/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-canister#dfx-canister-update-settings) command can be used with the `--add-controller <controller>` or `--remove-controller <controller>` flag to add or remove controllers respectively.
```
